### PR TITLE
PR: Add scrollbar to autosave dialog

### LINKED ISSUE #9125: Autosave recovery dialog expands beyond screen instead of using scrollbars with too many items
If I have more autosave files than will fit on my monitor (I have a 1200p one and just had it happen), then the dialog just expands outside of the visible screen (making it impossible to interact with the rest) rather than limiting its size and adding a scroll bar to the file list. Therefore, we should limit the size of the dialog, or the number of list items shown on screen accordingly to comfortably fit on a 720p monitor (or the current size, if practicable) and add a scroll bar widget to show the rest if that is the case. @jitseniesen take note, thanks!

### PR DESCRIPTION
**Pull request by juanis2112**
## Description of Changes

* [ ] Wrote at least one-line docstrings (for any new functions)
* [ ] Added unit test(s) covering the changes (if testable)

* [x] Included a screenshot or animation (if affecting the UI, see [Licecap](https://www.cockos.com/licecap/))

autosave

### Issue(s) Resolved

Fixes #9125

### Affirmation

By submitting this Pull Request or typing my (user)name below,
I affirm the [Developer Certificate of Origin](https://developercertificate.org)
with respect to all commits and content included in this PR,
and understand I am releasing the same under Spyder's MIT (Expat) license.

I certify the above statement is true and correct:
Juanis2112
dalthviz

### CODE REVIEW

**Comment by goanpeca** [comment]
@juanis2112 remember to add a screenshot 📷

File: spyder/plugins/editor/utils/autosave.py [hunk 1]
```diff
@@ -174,7 +174,8 @@ def try_recover_from_autosave(self):
         the pid files.
         """
         files_to_recover, pidfiles = self.get_files_to_recover()
-        dialog = RecoveryDialog(files_to_recover, parent=self.editor)
+        parent = self.editor if running_under_pytest() else self.editor.main
+        dialog = RecoveryDialog(files_to_recover, parent=parent)
```
Comment (jitseniesen) [inline spyder/plugins/editor/utils/autosave.py]: Out of curiosity, why this change?

Comment (dalthviz) [reply spyder/plugins/editor/utils/autosave.py]: The initial change was to use the `MainWindow` instead of the `Editor` to properly set the `styleSheet` for the Widget (before this the dialog always was shown using the light theme). However, when running the test suit we use a mock for the `MainWindow` that inherits from a 
`QObject`, and since the dialog needs a QWidget as parent, the tests suit was failing here. Any suggestions to handle this are greatly appreciated :)

File: spyder/plugins/editor/widgets/recover.py [hunk 2]
```diff
@@ -99,6 +99,8 @@ def __init__(self, autosave_mapping, parent=None):
         self.add_table()
         self.add_cancel_button()
         self.setWindowTitle(_('Recover from autosave'))
+        self.setFixedSize(670, 400)
```
Comment (goanpeca) [inline spyder/plugins/editor/widgets/recover.py]: Any particular reason for this fixed size? should we make it resizable?

File: spyder/plugins/editor/widgets/recover.py [hunk 3]
```diff
@@ -147,8 +149,8 @@ def add_table(self):
         table.setHorizontalHeaderLabels(labels)
         table.verticalHeader().hide()
 
-        table.setHorizontalScrollBarPolicy(Qt.ScrollBarAlwaysOff)
-        table.setVerticalScrollBarPolicy(Qt.ScrollBarAlwaysOff)
+        table.setHorizontalScrollBarPolicy(Qt.ScrollBarAsNeeded)
```
Comment (goanpeca) [inline spyder/plugins/editor/widgets/recover.py]: Should not this be off? by default?

File: spyder/plugins/editor/widgets/recover.py (hunk 2 above)
Comment (dalthviz) [reply spyder/plugins/editor/widgets/recover.py]: Basically I use this fixed size to show all the table horizontally and to prevent the dialog to get too tall, but maybe I can set the width to the width of the table instead of a fixed number. About the resize I think @ccordoba12 told me to prevent it

File: spyder/plugins/editor/widgets/recover.py (hunk 3 above)
Comment (dalthviz) [reply spyder/plugins/editor/widgets/recover.py]: Maybe if I set the width using the table width I can disable this, should I do it?

File: spyder/plugins/editor/widgets/recover.py (hunk 2 above)
Comment (dalthviz) [reply spyder/plugins/editor/widgets/recover.py]: Basically I use this fixed size to show all the table horizontally and to prevent the dialog to get too tall, but maybe I can set the width to the width of the table instead of a fixed number. About the resize I think @ccordoba12 told me to prevent it

File: spyder/plugins/editor/widgets/recover.py (hunk 3 above)
Comment (goanpeca) [reply spyder/plugins/editor/widgets/recover.py]: It is ok, we can merge as is!

File: spyder/plugins/editor/widgets/recover.py (hunk 2 above)
Comment (goanpeca) [reply spyder/plugins/editor/widgets/recover.py]: It is ok, we can merge as is!

File: spyder/plugins/editor/widgets/recover.py (hunk 3 above)
Comment (jitseniesen) [reply spyder/plugins/editor/widgets/recover.py]: > Maybe if I set the width using the table width I can disable this, should I do it?

The dialog can get arbitrarily wide if you have really long directory or file names, so I think it is dangerous to use the table width.

File: spyder/plugins/editor/utils/autosave.py (hunk 1 above)
Comment (jitseniesen) [reply spyder/plugins/editor/utils/autosave.py]: I see, I had this problem that the parent should be a `QWidget` before; unfortunately, I don't know of a way around this, so I'm afraid I can't help you.

File: spyder/plugins/editor/widgets/recover.py [hunk 4]
```diff
@@ -206,6 +202,13 @@ def add_cancel_button(self):
         button_box.rejected.connect(self.reject)
         self.layout.addWidget(button_box)
 
+    def center(self):
```
Comment (ccordoba12) [inline spyder/plugins/editor/widgets/recover.py]: @dalthviz, what happens when we try to center this and you have two monitors in your system?

Comment (dalthviz) [reply spyder/plugins/editor/widgets/recover.py]: The dialog is shown centered in the first screen

Comment (dalthviz) [reply spyder/plugins/editor/widgets/recover.py]: Maybe instead of `QApplication.desktop().screenGeometry(0)` we should use `QApplication.primaryScreen()` here @ccordoba12 ?

File: spyder/plugins/editor/utils/autosave.py (hunk 1 above)
Comment (dalthviz) [reply spyder/plugins/editor/utils/autosave.py]: No problem, thanks for looking into this :+1:

File: spyder/plugins/editor/widgets/recover.py (hunk 4 above)
Comment (ccordoba12) [reply spyder/plugins/editor/widgets/recover.py]: `primaryScreen` is the screen where Spyder was launched?

Comment (goanpeca) [reply spyder/plugins/editor/widgets/recover.py]: Probably not, it is what the system defines as "primary" so to know which screen the widget lives (The QMainWindow for example) we could look into:

https://stackoverflow.com/questions/49167204/find-screen-a-qwidget-is-located-on

Comment (goanpeca) [reply spyder/plugins/editor/widgets/recover.py]: Probably not, it is what the system defines as "primary" so to know which screen the widget lives (The QMainWindow for example) we could look into:

https://stackoverflow.com/questions/49167204/find-screen-a-qwidget-is-located-on

Comment (dalthviz) [reply spyder/plugins/editor/widgets/recover.py]: At the end this resulted in using the window settings from the config files. This is a preview:

auto

File: spyder/app/mainwindow.py [hunk 5]
```diff
@@ -1522,6 +1522,19 @@ def load_window_settings(self, prefix, default=False, section='main'):
         return hexstate, window_size, prefs_dialog_size, pos, is_maximized, \
                is_fullscreen
 
+    def get_window_center(self, load=False):
```
Comment (ccordoba12) [inline spyder/app/mainwindow.py]: @dalthviz, I think you're assuming here that the window was maximized the previous time it was closed, which is normally the case. But what would happen if it was closed with half the size of the screen?

Comment (dalthviz) [reply spyder/app/mainwindow.py]: It will appear centered relative to the last position of the MainWindow:

autod

Comment (ccordoba12) [reply spyder/app/mainwindow.py]: Mmmh, I don't like this too much. How is the splash screen correctly centered?

Comment (dalthviz) [reply spyder/app/mainwindow.py]: The splash screen is always centered relative to the first screen (it doesn't depend on where the MainWindow will appear)

Comment (ccordoba12) [reply spyder/app/mainwindow.py]: Then let's do the same for the Autosave dialog. I think it's the best we can do.
